operator/cluster/clusterlogic.go

```diff
@@ -279,26 +296,18 @@ func Scale(clientset *kubernetes.Clientset, client *rest.RESTClient, replica *cr
 
 	cluster.Spec.UserLabels[config.LABEL_DEPLOYMENT_NAME] = replica.Spec.Name
 
-	// iterate through all of the tablespaces and attempt to create their PVCs
-	// for the replcia
-	for tablespaceName, storageSpec := range cluster.Spec.TablespaceMounts {
-		// attempt to create the tablespace PVC. If it fails to create, log the
-		// error and publish the failure event
-		// Note that we specify **replica.Spec.Name** in order to create distinct
-		// PVCs for this replica, but we use the **cluster.Spec.ClusterName** for the
-		// "pgcluster" Label
-		tablespacePVCName := operator.GetTablespacePVCName(replica.Spec.Name, tablespaceName)
-
-		if err := CreateTablespacePVC(clientset, namespace, cluster.Spec.ClusterName, tablespacePVCName, &storageSpec); err != nil {
-			log.Error(err)
-			publishScaleError(namespace, replica.ObjectMeta.Labels[config.LABEL_PGOUSER], cluster)
-			return err
-		}
-	}
-
 	// set up a map of the names of the tablespaces as well as the storage classes
 	tablespaceStorageTypeMap := operator.GetTablespaceStorageTypeMap(cluster.Spec.TablespaceMounts)
 
+	// combine supplemental groups from all volumes
+	var supplementalGroups []int64
+	{
+		supplementalGroups = append(supplementalGroups, dataVolume.SupplementalGroups...)
+		for _, v := range tablespaceVolumes {
+			supplementalGroups = append(supplementalGroups, v.SupplementalGroups...)
+		}
+	}
```

Comment:
I see this pattern enough, perhaps we just turn this into a function?

Reply:
I think the right way forward is to spread it out rather than consolidate it. e.g. I would like tablespace code to modify a pod spec for tablespaces.

Reply:
Makes sense. Looking forward to more discussions